sway-ir/Cargo.toml

```diff
@@ -0,0 +1,17 @@
+[package]
```

Comment:
IMO you should add a `publish = false` here, unless we're actually planning on publishing this binary.

Reply:
Do you mean the whole library?  In this case, yeah we're planning on publishing it, much like `sway-types`.

But if you mean the `opt` binary, then yeah, maybe we shouldn't expose that... or maybe give it a better name?  `cargo install sway-ir` would also put `opt` in `~/.cargo/bin` I assume.  Hrmm.

Reply:
Can you turn `opt` into `forc` command? Similar to how `forc lsp` runs `sway-server`?

Reply:
`opt` is a binary interface to the IR library specifically, and doesn't necessarily have anything to do with Sway.  It takes some IR in, performs passes and spits more IR out.  It's not really for Sway users, more for us developers.

Based on a similar [LLVM tool](https://llvm.org/docs/CommandGuide/opt.html).

Reply:
If opt is just an internal binary, I say we can just leave it as is and don’t document it. It could be useful for testing

Reply:
Yeah, we're definitely keeping it somehow, it'll actually be the basis for all integration testing.

So, if `sway-ir` is installed as a library dependency, then `opt` wouldn't be installed to `~/.cargo/bin`?  But if it was installed explicitly, i.e., `cargo install sway-ir` then it would?  I think it should probably be renamed.

Reply:
Can you separate out the library from the binary? Then we can publish the library but not the binary.

Reply:
Ahh, no, I'm wrong I think -- binaries are only installed from binary crates.  For library crates with binaries they don't get installed, AFAICT.  All good.